In [1]:
import pandas as pd
import os
from os import listdir
import numpy as np

In [31]:
filelist=os.popen('ls *.csv').read()
files=filelist.split('\n')
files=files[:-1]

In [46]:
## summarize all the test
df = pd.read_csv(files[0],index_col=0)
df.index = df.Algorithm
df=df.drop(['Algorithm','test','CrossValMeans','CrossValerrors'],axis=1)
df.columns=[files[0][:-4]]
for file in files[1:]:
    data = pd.read_csv(file,index_col=0)
    data.index = data.Algorithm
    data=data.drop(['Algorithm','test','CrossValMeans','CrossValerrors'],axis=1)
    data.columns=[file[:-4]]
    df = pd.concat([df,data],axis=1)

In [48]:
df = df.T
df = df.round(3)

In [50]:
df.to_csv('../time.csv')

In [86]:
## summarize others
df = pd.read_csv(files[0],index_col=0)
df.index = df.Algorithm
df=df.drop(['Algorithm','Time'],axis=1)
df.columns=[files[0][:-4] + '_1', files[0][:-4] + '_2', files[0][:-4] + '_3']
for file in files[1:]: 
    data = pd.read_csv(file,index_col=0)
    data.index = data.Algorithm
    data=data.drop(['Algorithm','Time'],axis=1)
    data.columns=[file[:-4] + '_1', file[:-4] + '_2', file[:-4] + '_3']
    df = pd.concat([df,data],axis=1)

In [87]:
df = pd.read_csv(files[0],index_col=0)
df.index = df.Algorithm
df=df.drop(['Algorithm','Time'],axis=1).round(2)
df['valid'] = df['CrossValMeans'].map(str)+ 'A1O2' + df['CrossValerrors'].map(str)
df = df.drop(['CrossValMeans','CrossValerrors'],axis=1)[['valid','test']]
df.columns=[files[0][:-4] + '训练集精度', files[0][:-4] + '测试集精度']
for file in files[1:]: 
    data = pd.read_csv(file,index_col=0)
    data.index = data.Algorithm
    data=data.drop(['Algorithm','Time'],axis=1).round(2)
    data['valid'] = data['CrossValMeans'].map(str)+ 'A1O2' + data['CrossValerrors'].map(str)
    data = data.drop(['CrossValMeans','CrossValerrors'],axis=1)[['valid','test']]
    data.columns=[file[:-4] + '训练集精度', file[:-4] + '测试集精度']
    df = pd.concat([df,data],axis=1)

In [89]:
df = df.T
df.to_csv('../accuracy.csv')

In [90]:
df

Algorithm         DecisionTree           SVC LogisticRegression   KNeighboors  \
abalone训练集精度      0.27A1O20.02  0.28A1O20.01       0.27A1O20.02  0.27A1O20.01   
abalone测试集精度              0.27          0.25               0.26          0.24   
adult训练集精度        0.81A1O20.01  0.85A1O20.01       0.82A1O20.01  0.83A1O20.01   
adult测试集精度                0.81          0.85               0.83          0.83   
arcene训练集精度        0.7A1O20.11  0.75A1O20.11       0.76A1O20.09   0.81A1O20.1   
arcene测试集精度               0.71          0.76               0.84           0.8   
australia训练集精度    0.87A1O20.02  0.89A1O20.03       0.87A1O20.03  0.88A1O20.03   
australia测试集精度            0.81          0.82               0.86          0.85   
banknote训练集精度      0.99A1O20.0    1.0A1O20.0       0.99A1O20.01    1.0A1O20.0   
banknote测试集精度                1             1               0.99             1   
bioresponse训练集精度  0.72A1O20.03  0.77A1O20.03       0.73A1O20.03  0.75A1O20.02   
bioresponse测试集精度          0.75          0.78               0.73          0.74   
breast训练集精度       0.97A1O20.04  0.97A1O20.03       0.96A1O20.03  0.98A1O20.02   
breast测试集精度               0.97          0.97               0.95          0.95   
churn训练集精度         0.93A1O20.0  0.92A1O20.01       0.86A1O20.01   0.89A1O20.0   
churn测试集精度                0.93          0.91               0.87           0.9   
credit训练集精度       0.73A1O20.01  0.82A1O20.01        0.81A1O20.0  0.79A1O20.01   
credit测试集精度               0.73          0.81                0.8          0.79   
dermatology训练集精度  0.95A1O20.03  0.97A1O20.02       0.97A1O20.02  0.97A1O20.02   
dermatology测试集精度          0.93          0.98               0.98          0.96   
diabetes训练集精度     0.72A1O20.01  0.76A1O20.03       0.76A1O20.02  0.77A1O20.01   
diabetes测试集精度             0.77          0.78               0.81          0.76   
digit训练集精度        0.85A1O20.01   0.96A1O20.0        0.91A1O20.0   0.94A1O20.0   
digit测试集精度                0.85          0.96               0.91          0.94   
glass训练集精度        0.67A1O20.08   0.7A1O20.07       0.65A1O20.05  0.71A1O20.07   
glass测试集精度                0.58          0.69               0.64           0.7   
heart训练集精度        0.79A1O20.06  0.83A1O20.05       0.83A1O20.05  0.84A1O20.05   
heart测试集精度                0.75          0.83               0.82          0.82   
internetad训练集精度   0.96A1O20.01  0.96A1O20.01       0.96A1O20.01  0.95A1O20.02   
internetad测试集精度           0.96          0.97               0.96          0.95   
iris训练集精度         0.94A1O20.04  0.97A1O20.04       0.96A1O20.04  0.97A1O20.02   
iris测试集精度                 0.94          0.95               0.96          0.94   
micromass训练集精度    0.79A1O20.05  0.51A1O20.05       0.89A1O20.03  0.59A1O20.07   
micromass测试集精度            0.76          0.42                0.8          0.58   
mushroom训练集精度       1.0A1O20.0    1.0A1O20.0        0.97A1O20.0    1.0A1O20.0   
mushroom测试集精度                1             1               0.97             1   
nursery训练集精度       0.99A1O20.0    1.0A1O20.0       0.76A1O20.01   0.94A1O20.0   
nursery测试集精度              0.99             1               0.77          0.96   
splice训练集精度       0.88A1O20.02   0.84A1O20.0       0.81A1O20.01  0.79A1O20.02   
splice测试集精度               0.89          0.85               0.83          0.82   
thyroid训练集精度      0.69A1O20.02  0.71A1O20.02       0.71A1O20.02  0.69A1O20.02   
thyroid测试集精度              0.68          0.71               0.71          0.71   
vehicle训练集精度       0.7A1O20.02  0.82A1O20.02        0.8A1O20.02  0.72A1O20.04   
vehicle测试集精度              0.67          0.86               0.79          0.67   
wine训练集精度         0.59A1O20.03  0.63A1O20.02       0.58A1O20.04   0.6A1O20.04   
wine测试集精度                 0.64          0.62               0.59           0.6   

Algorithm        MultipleLayerPerceptron    NaiveBayes  RandomForest  \
abalone训练集精度                0.28A1O20.01  0.24A1O20.02  0.28A1O20.02   
abalone测试集精度                        0.26        

In [113]:
### overfit analysis
df = pd.read_csv(files[0],index_col=0)
df.index = df.Algorithm
df['overfit'] = df['CrossValMeans'] >  df['test']
df = df[['overfit']]
for file in files[1:]: 
    data = pd.read_csv(file,index_col=0)
    data.index = data.Algorithm
    data['overfit'] = data['CrossValMeans']  >  data['test']
    data = data[['overfit']]
    df = pd.concat([df,data],axis=1)

In [116]:
5/23

0.21739130434782608

In [121]:
overfit=df.mean(axis=1).round(3)
overfit[7] = 0.217
overfit

Algorithm
DecisionTree               0.478
SVC                        0.435
LogisticRegression         0.435
KNeighboors                0.739
MultipleLayerPerceptron    0.565
NaiveBayes                 0.435
RandomForest               0.478
AdaBoost                   0.217
dtype: float64

In [125]:
overfit.to_csv('../overfit.csv',header=True)

## High dimension

In [30]:
data = pd.read_csv('arcene.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors      Time      test
0             DecisionTree       0.703787        0.107101  1.281386  0.707129
1                      SVC       0.752136        0.108303  1.340125  0.757426
2       LogisticRegression       0.763742        0.092641  1.615689  0.844554
3              KNeighboors       0.808154        0.097384  0.550664  0.798614
4  MultipleLayerPerceptron       0.760116        0.101103  6.754241  0.747327
5               NaiveBayes       0.554663        0.115142  0.380704  0.546139
6             RandomForest       0.762630        0.101794  0.464299  0.778614
7                 AdaBoost       0.701660        0.108003  1.284511  0.701980

**train_number    2459.000000
test_number      820.000000
classes            2.000000
features        1558.000000
binary             0.998074
categorical        0.000000
continous          0.001926
meancor            0.040399
outliers           0.010776
classentropy       0.584191
dimension          0.475145**

In [6]:
data = pd.read_csv('internetad.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors       Time      test
0             DecisionTree       0.961793        0.013087   5.939167  0.963415
1                      SVC       0.962201        0.012281  19.339123  0.967073
2       LogisticRegression       0.964232        0.014129  13.420060  0.963415
3              KNeighboors       0.952045        0.016741   7.130864  0.950000
4  MultipleLayerPerceptron       0.966681        0.015783  40.720596  0.964634
5               NaiveBayes       0.787308        0.018222   0.779171  0.779268
6             RandomForest       0.969920        0.015403   1.444854  0.980488
7                 AdaBoost       0.961793        0.016219  18.123823  0.969512

RF is the best, except for NB, all classifier behave similarly, 99% of the data is binary, and class is a little skewed, high dimension. 

**train_number    2813.000000
test_number      938.000000
classes            2.000000
features        1776.000000
binary             0.470158
categorical        0.417230
continous          0.112613
meancor            0.095606
outliers           0.015740
classentropy       0.994842
dimension          0.473474**

In [7]:
data = pd.read_csv('bioresponse.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors       Time      test
0             DecisionTree       0.718799        0.028363   6.240876  0.745203
1                      SVC       0.771076        0.029711  54.459361  0.775053
2       LogisticRegression       0.726968        0.028033  31.002467  0.733475
3              KNeighboors       0.750092        0.020989  14.270545  0.744136
4  MultipleLayerPerceptron       0.758262        0.028624  71.477207  0.767591
5               NaiveBayes       0.605396        0.021591   1.040884  0.565032
6             RandomForest       0.762893        0.025287   1.239404  0.799574
7                 AdaBoost       0.708484        0.027577   4.347092  0.724947

SVM, RF is the best, most are discrete features

**train_number     428.000000
test_number      143.000000
classes           20.000000
features        1300.000000
binary             0.030769
categorical        0.349231
continous          0.620000
meancor            0.192539
outliers           0.014910
classentropy       0.971764
dimension          2.276708**

In [8]:
data = pd.read_csv('micromass.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors       Time      test
0             DecisionTree       0.786057        0.052659   0.334813  0.755245
1                      SVC       0.508543        0.050301   2.254210  0.419580
2       LogisticRegression       0.885139        0.027279  11.985113  0.797203
3              KNeighboors       0.592936        0.067950   0.357981  0.580420
4  MultipleLayerPerceptron       0.867042        0.052934   6.570259  0.804196
5               NaiveBayes       0.713681        0.064792   0.170112  0.699301
6             RandomForest       0.802764        0.080051   0.218039  0.783217
7                 AdaBoost       0.782791        0.061121   0.345133  0.790210

MLP(best), LR, RF, Ada are great, SVM and KNN are even poorer than NB.  
LR takes longest time.
Super high dimension, size < features, 62% continuous data, 0.2 corr.

Maybe there are many irrevelant features, therefore SVM and KNN have poor performance.

## multiclass

train_number    31500.000000
test_number     10500.000000
**classes            10.000000**
features          708.000000
binary              0.026836
categorical         0.100282
**continous           0.872881**
meancor             0.078436
outliers            0.011241
classentropy        0.999381
dimension           0.016857

In [9]:
data = pd.read_csv('digit.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors         Time  \
0             DecisionTree       0.850953        0.007497    38.496567   
1                      SVC       0.956189        0.003465   989.778891   
2       LogisticRegression       0.907173        0.004502  1482.803813   
3              KNeighboors       0.937365        0.004390   700.641372   
4  MultipleLayerPerceptron       0.966158        0.002100   123.627193   
5               NaiveBayes       0.528064        0.008095     6.855807   
6             RandomForest       0.935143        0.005330    11.553082   
7                 AdaBoost       0.851873        0.007886    37.507837   

       test  
0  0.846857  
1  0.957429  
2  0.905905  
3  0.936952  
4  0.967714  
5  0.528095  
6  0.933429  
7  0.845619

(1)MLP is the best, since each feature contribute no information to the class and the data is continuous. 
(2)AdaBoost  = DT, the running time of DT is more than RF and Ada, because DT is complex here.
(3)LR slow, 

train_number    3132.000000
test_number     1045.000000
**classes           28.000000**
features           8.000000
binary             0.000000
categorical        0.125000
continous          0.875000
meancor                 NaN
outliers           0.004720
classentropy       0.749270
dimension          0.001915

In [10]:
data = pd.read_csv('abalone.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors       Time      test
0             DecisionTree       0.261765        0.020220   0.038715  0.252632
1                      SVC       0.280313        0.022903   9.647782  0.249761
2       LogisticRegression       0.272549        0.031626   1.301132  0.259330
3              KNeighboors       0.274218        0.025982   0.090781  0.235407
4  MultipleLayerPerceptron       0.279731        0.021577  22.984903  0.254545
5               NaiveBayes       0.244574        0.029464   0.055922  0.211483
6             RandomForest       0.280715        0.025083   0.849657  0.241148
7                 AdaBoost       0.207370        0.022425   0.077349  0.192344

MLP, RF, SVM, LR great. AdaBoost poor. Because the base learner perform poor.

train_number    1199.000000
test_number      400.000000
**classes            6.000000**
features          11.000000
binary             0.000000
categorical        0.000000
**continous          1.000000**
meancor            0.267784
outliers           0.010347
**classentropy       0.661155**
dimension          0.006879

In [11]:
data = pd.read_csv('wine.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors       Time    test
0             DecisionTree       0.595440        0.046487   0.025946  0.6075
1                      SVC       0.634784        0.039702   2.112117  0.6175
2       LogisticRegression       0.579754        0.051518   0.177886  0.5875
3              KNeighboors       0.598893        0.028087   0.051198  0.5925
4  MultipleLayerPerceptron       0.614077        0.048338  16.451973  0.6025
5               NaiveBayes       0.537317        0.045358   0.030639  0.5400
6             RandomForest       0.674763        0.036932   0.967383  0.6800
7                 AdaBoost       0.593337        0.065930   0.042797  0.6275

RF is the best, DT = Ad

train_number    160.000000
test_number      54.000000
classes           6.000000
features          9.000000
binary            0.000000
categorical       0.000000
continous         1.000000
meancor           0.382727
outliers          0.017134
classentropy      0.841998
dimension         0.042056

In [12]:
data = pd.read_csv('glass.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors      Time      test
0             DecisionTree       0.723886        0.178835  0.019522  0.685185
1                      SVC       0.700170        0.083986  0.048421  0.740741
2       LogisticRegression       0.681904        0.056903  0.303863  0.592593
3              KNeighboors       0.720641        0.095739  0.025382  0.685185
4  MultipleLayerPerceptron       0.717425        0.087929  3.008001  0.759259
5               NaiveBayes       0.512606        0.164192  0.035800  0.629630
6             RandomForest       0.772314        0.070489  0.532150  0.777778
7                 AdaBoost       0.673283        0.100884  0.025040  0.685185

RF is the best, DT = Ad

train_number    274.000000
test_number      92.000000
classes           6.000000
features         34.000000
binary            0.029412
categorical       0.941176
continous         0.029412
meancor           0.150507
outliers          0.000000
classentropy      0.941057
dimension         0.092896

In [14]:
data = pd.read_csv('dermatology.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors      Time      test
0             DecisionTree       0.956439        0.039636  0.020653  0.967391
1                      SVC       0.971460        0.031184  0.090327  0.978261
2       LogisticRegression       0.974908        0.032226  0.093921  0.989130
3              KNeighboors       0.971583        0.026428  0.028010  0.967391
4  MultipleLayerPerceptron       0.971062        0.031283  1.380750  0.989130
5               NaiveBayes       0.872493        0.038025  0.024771  0.858696
6             RandomForest       0.985021        0.024736  0.222214  0.989130
7                 AdaBoost       0.944167        0.038326  0.029070  0.956522

RF, LR, MLP best, DT = Ad

## Large size data

train_number    36631.000000
test_number     12211.000000
classes             2.000000
features           14.000000
binary              0.071429
categorical         0.500000
continous           0.428571
meancor             0.047522
outliers            0.019048
classentropy        0.793844
dimension           0.000287

In [16]:
data = pd.read_csv('adult.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors        Time  \
0             DecisionTree       0.811963        0.006127    1.099756   
1                      SVC       0.848925        0.006018  101.689901   
2       LogisticRegression       0.823564        0.006248    0.793124   
3              KNeighboors       0.829980        0.005255   20.056257   
4  MultipleLayerPerceptron       0.850018        0.007470   81.654333   
5               NaiveBayes       0.804892        0.005196    0.412078   
6             RandomForest       0.850919        0.004026    1.800816   
7                 AdaBoost       0.828862        0.008026   32.940332   

       test  
0  0.811727  
1  0.853411  
2  0.829089  
3  0.829170  
4  0.850872  
5  0.803783  
6  0.850954  
7  0.827369

MLP, RF, SVM best; Ada >= DT

train_number    22500.000000
test_number      7500.000000
classes             2.000000
features           23.000000
binary              0.043478
categorical         0.347826
continous           0.608696
meancor             0.200964
outliers            0.015662
classentropy        0.762353
dimension           0.000767

In [17]:
data = pd.read_csv('credit.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors       Time      test
0             DecisionTree       0.728310        0.009321   3.606902  0.725200
1                      SVC       0.820488        0.005207  82.099281  0.814400
2       LogisticRegression       0.812045        0.002171   0.788667  0.803333
3              KNeighboors       0.792354        0.005936  14.930116  0.786267
4  MultipleLayerPerceptron       0.814089        0.004594  53.467542  0.810933
5               NaiveBayes       0.669822        0.031445   0.151334  0.663733
6             RandomForest       0.807466        0.004164   2.324265  0.800533
7                 AdaBoost       0.785466        0.006868  18.395862  0.774800

SVM, MLP best; Ada > DT, DT is poor 

train_number    9720.000000
test_number     3240.000000
classes            5.000000
features           8.000000
binary             0.125000
categorical        0.875000
continous          0.000000
meancor            0.048074
outliers                NaN
classentropy       0.739255
dimension          0.000617

In [18]:
data = pd.read_csv('nursery.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors       Time      test
0             DecisionTree       0.994547        0.002949   0.198281  0.994136
1                      SVC       0.996296        0.001852  33.118687  0.997531
2       LogisticRegression       0.763776        0.015008   0.178434  0.765741
3              KNeighboors       0.947016        0.005435   0.547293  0.961111
4  MultipleLayerPerceptron       0.999897        0.000308  55.114204  0.999383
5               NaiveBayes       0.639087        0.014958   0.053256  0.641049
6             RandomForest       0.994959        0.001977   2.925033  0.995679
7                 AdaBoost       0.994032        0.002754   0.167261  0.993827

LR poorly, KNN is modest,  others high accuracy

## High correlation

train_number    426.000000
test_number     143.000000
classes           2.000000
features         30.000000
binary            0.000000
categorical       0.000000
continous         1.000000
meancor           0.319730
outliers          0.012361
classentropy      0.952635
dimension         0.052724

In [19]:
data = pd.read_csv('breast.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors      Time      test
0             DecisionTree       0.957473        0.036322  0.021802  0.916084
1                      SVC       0.976462        0.021063  0.059527  0.972028
2       LogisticRegression       0.978790        0.012786  0.034617  0.965035
3              KNeighboors       0.964560        0.032407  0.031993  0.972028
4  MultipleLayerPerceptron       0.978788        0.022218  1.566920  0.979021
5               NaiveBayes       0.927011        0.034638  0.025976  0.930070
6             RandomForest       0.971592        0.027797  0.669910  0.916084
7                 AdaBoost       0.948382        0.033228  0.028091  0.902098

RF and DT overfit, others are great. NB is superisingly great.

train_number    1029.000000
test_number      343.000000
classes            2.000000
features           4.000000
binary             0.000000
categorical        0.000000
continous          1.000000
meancor            0.446389
outliers           0.006560
classentropy       0.991128
dimension          0.002915

In [21]:
data = pd.read_csv('banknote.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors      Time      test
0             DecisionTree       0.987387        0.011485  0.022732  0.991254
1                      SVC       1.000000        0.000000  0.081137  1.000000
2       LogisticRegression       0.985483        0.011571  0.049898  0.997085
3              KNeighboors       0.998068        0.003865  0.031564  1.000000
4  MultipleLayerPerceptron       1.000000        0.000000  2.162261  1.000000
5               NaiveBayes       0.834836        0.043620  0.024941  0.845481
6             RandomForest       0.992242        0.007264  0.570577  0.994169
7                 AdaBoost       0.987415        0.013002  0.028408  0.988338

NB poor

train_number    112.000000
test_number      38.000000
classes           3.000000
features          4.000000
binary            0.000000
categorical       0.000000
continous         1.000000
meancor           0.486321
outliers          0.001667
classentropy      1.000000
dimension         0.026667

In [22]:
data = pd.read_csv('iris.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors      Time      test
0             DecisionTree       0.965035        0.042994  0.017905  0.947368
1                      SVC       0.974126        0.056816  0.026627  0.947368
2       LogisticRegression       0.971818        0.043112  0.083487  0.947368
3              KNeighboors       0.983217        0.033712  0.028999  0.947368
4  MultipleLayerPerceptron       0.972727        0.041660  0.958430  0.947368
5               NaiveBayes       0.945944        0.060236  0.019318  0.973684
6             RandomForest       0.965035        0.059173  0.204246  0.973684
7                 AdaBoost       0.955035        0.045268  0.023105  0.973684

test number is too small

train_number    634.000000
test_number     212.000000
classes           4.000000
features         18.000000
binary            0.000000
categorical       0.055556
continous         0.944444
meancor           0.454220
outliers          0.002990
classentropy      0.999533
dimension         0.021277

In [23]:
data = pd.read_csv('vehicle.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors      Time      test
0             DecisionTree       0.717704        0.047241  0.026921  0.702830
1                      SVC       0.843637        0.023886  0.303040  0.834906
2       LogisticRegression       0.805961        0.044972  0.302055  0.787736
3              KNeighboors       0.734820        0.043853  0.034577  0.655660
4  MultipleLayerPerceptron       0.854900        0.031311  6.572449  0.811321
5               NaiveBayes       0.444889        0.044754  0.027133  0.466981
6             RandomForest       0.758634        0.034055  0.677931  0.716981
7                 AdaBoost       0.687911        0.057003  0.044036  0.702830

SVC is the best; tree is poor; RF,KNN and MLP overfit

## Low dimension < 0.005
['abalone', 'adult', 'banknote', 'churn', 'credit', 'mushroom','nursery'],

In [24]:
data = pd.read_csv('churn.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors       Time    test
0             DecisionTree       0.944805        0.012713   0.194600  0.9512
1                      SVC       0.925338        0.012984   9.104529  0.9376
2       LogisticRegression       0.864547        0.015946   0.088513  0.8744
3              KNeighboors       0.893330        0.009501   0.382140  0.8984
4  MultipleLayerPerceptron       0.926668        0.011069  31.448615  0.9328
5               NaiveBayes       0.872276        0.012148   0.045634  0.8680
6             RandomForest       0.960272        0.008697   3.179116  0.9624
7                 AdaBoost       0.922406        0.011049   0.255650  0.9328

In [25]:
data = pd.read_csv('mushroom.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors      Time      test
0             DecisionTree       1.000000        0.000000  0.073558  1.000000
1                      SVC       1.000000        0.000000  5.799253  1.000000
2       LogisticRegression       0.965535        0.006224  0.202595  0.967996
3              KNeighboors       1.000000        0.000000  0.689679  1.000000
4  MultipleLayerPerceptron       1.000000        0.000000  4.792615  1.000000
5               NaiveBayes       0.922203        0.005963  0.066466  0.933038
6             RandomForest       1.000000        0.000000  0.132852  1.000000
7                 AdaBoost       1.000000        0.000000  0.078253  1.000000

In [26]:
data = pd.read_csv('thyroid.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors       Time      test
0             DecisionTree       0.685789        0.024406   0.038348  0.688571
1                      SVC       0.712041        0.024172   3.845595  0.702857
2       LogisticRegression       0.708241        0.026242   0.118715  0.714286
3              KNeighboors       0.697295        0.025681   0.148629  0.708571
4  MultipleLayerPerceptron       0.672000        0.018792  19.111880  0.688571
5               NaiveBayes       0.176761        0.027131   0.037410  0.180000
6             RandomForest       0.718210        0.021637   0.904770  0.702857
7                 AdaBoost       0.612386        0.027889   0.091245  0.624286

## Binary classification
['adult', 'australia', 'banknote', 'bioresponse', 'breast', 'churn','credit', 'diabetes', 'heart', 'internetad', 'mushroom']

In [27]:
data = pd.read_csv('australia.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors      Time      test
0             DecisionTree       0.885860        0.045197  0.020114  0.809249
1                      SVC       0.891742        0.046467  0.145858  0.832370
2       LogisticRegression       0.874359        0.048188  0.031018  0.855491
3              KNeighboors       0.874359        0.038460  0.031812  0.855491
4  MultipleLayerPerceptron       0.884050        0.040990  5.056243  0.791908
5               NaiveBayes       0.849246        0.049723  0.024456  0.878613
6             RandomForest       0.891817        0.051585  0.264628  0.861272
7                 AdaBoost       0.851056        0.053121  0.033520  0.745665

MLP,DT overfit, NB best

In [29]:
data = pd.read_csv('heart.csv',index_col=0)
data

Algorithm  CrossValMeans  CrossValerrors      Time      test
0             DecisionTree       0.852536        0.112491  0.026101  0.776316
1                      SVC       0.850264        0.044501  0.052899  0.828947
2       LogisticRegression       0.833416        0.047788  0.037462  0.842105
3              KNeighboors       0.842523        0.059983  0.029357  0.750000
4  MultipleLayerPerceptron       0.837385        0.061389  1.332322  0.842105
5               NaiveBayes       0.829068        0.060501  0.023047  0.776316
6             RandomForest       0.851581        0.073310  0.224223  0.828947
7                 AdaBoost       0.794433        0.089266  0.024875  0.644737

In [ ]:
DT，KNN overfit, 